spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -66,6 +67,93 @@ object DistriOptimizer {
     optimMethod: OptimMethod[T]
   )
 
+  private def collectGlobalData[T: ClassTag](
+    models: RDD[Cache[T]],
+    parameters: AllReduceParameter[T],
+    operations: Array[String],
+    numFinishedModelUpdatesT: T,
+    subModelNumber: Int,
+    metrics: Metrics,
+    lookupDicBroadcast: Broadcast[Array[mutable.HashMap[Int, (Int, Int)]]])
+    (implicit ev: TensorNumeric[T]):
+    (Boolean, Double, Broadcast[Array[(Int, (Double, Double))]]) = {
+    var aggregatedG: Boolean = false
+    // array element is a tuple, first element is layer id, value is (wNorm2, gNorm2)
+    var gwNorm2ListBroadcast: Broadcast[Array[(Int, (Double, Double))]] = null
+    var l2Norm = 0.0
+    operations.foreach { oper =>
+      if (oper.equals("norm2Clip")) {
```

Comment:
so if we add another string, we also need to change the ```if else``` blocks? Why not wrap them with class...